cmd/controller/controller.go

```diff
@@ -303,3 +294,16 @@ func (c *Controller) Rotate(content []byte) ([]byte, error) {
 		return nil, fmt.Errorf("Unexpected resoure type: %s", s.GetObjectKind().GroupVersionKind().String())
 	}
 }
+
+func (c *Controller) attemptUnseal(ss *ssv1alpha1.SealedSecret) (*apiv1.Secret, string, error) {
+	return attemptUnseal(ss, c.keyRegistry)
+}
+
+func attemptUnseal(ss *ssv1alpha1.SealedSecret, keyRegistry *KeyRegistry) (*apiv1.Secret, string, error) {
+	for _, privKey := range keyRegistry.keys {
+		if secret, err := ss.Unseal(scheme.Codecs, privKey); err == nil {
+			return secret, "", nil
```

Comment:
Yes, that is correct.
The idea was to return the keyName that successfully decrypted a sealed secret. The check above was to avoid re-encryption with the same key a sealed secret was encrypted with. Fixed now.